### PR TITLE
Safekeeper metrics tests

### PR DESCRIPTION
**Pull request by yeputons**
This PR ensures that LSN guages in Safekeeper are reported and satisfy some basic requirements. Part of #541

WIP:

* [x] Take a look through #742 for better data retrieval from the Compute Node and stuff
* [x] Resolve conflicts with #742 
* [x] Understand what is the relation between `flush_lsn`, `commit_lsn` and `last_valid_lsn`: https://discord.com/channels/869525774699462656/869526396878323733/897550905090637855
* [x] Add actual assertions instead of debug output

### CODE REVIEW

File: test_runner/batch_others/test_wal_acceptor.py [hunk 1]
```diff
@@ -50,21 +61,103 @@ def test_many_timelines(zenith_env_builder: ZenithEnvBuilder):
...
+                    m.commit_lsns.append(sk_m.commit_lsn_inexact[timeline_id])
+
+                for flush_lsn, commit_lsn in zip(m.flush_lsns, m.commit_lsns):
+                    # Invariant. May be < when transaction is in progress.
+                    assert commit_lsn <= flush_lsn
+                # We only call collect_metrics() after a transaction is confirmed by
+                # the compute node, which only happens after a consensus of safekeepers
+                # has confirmed the transaction. In local tests it's quite often
+                # that all safekeepers has confirmed the transaction, so we ensure that
+                # until it's flaky.
```
Comment (arssher) [inline test_runner/batch_others/test_wal_acceptor.py:101]: Uh, I wouldn't leave it like that, I can't predict 'flakiness' degree here with enough precision.

File: test_runner/batch_others/test_wal_acceptor.py [hunk 2]
```diff
@@ -50,21 +61,103 @@ def test_many_timelines(zenith_env_builder: ZenithEnvBuilder):
...
+                    name=branch_detail["name"],
+                    latest_valid_lsn=branch_detail["latest_valid_lsn"],
+                )
+                for sk_m in sk_metrics:
+                    m.flush_lsns.append(sk_m.flush_lsn_inexact[timeline_id])
+                    m.commit_lsns.append(sk_m.commit_lsn_inexact[timeline_id])
+
+                for flush_lsn, commit_lsn in zip(m.flush_lsns, m.commit_lsns):
+                    # Invariant. May be < when transaction is in progress.
+                    assert commit_lsn <= flush_lsn
```
Comment (arssher) [inline test_runner/batch_others/test_wal_acceptor.py:96]: Will trigger of these assertions fail the test, if they happen from another thread?

File: test_runner/batch_others/test_wal_acceptor.py [hunk 3]
```diff
@@ -47,21 +57,74 @@ def test_many_timelines(zenith_cli, pageserver: ZenithPageserver, postgres: Post
...
-    # Check data
-    for pg in pgs:
+    # Check data for 2/3 branches
+    for pg in pgs[:-1]:
         res = pg.safe_psql("SELECT sum(key) FROM t")
         assert res[0] == (5000050000, )
 
+    final_branch_metrics = collect_metrics("after SELECT")
+    # Assume that LSNs (a) behave similarly in all branches; and (b) INSERT INTO alters LSN significantly.
+    # Also assume that safekeepers will not be significantly out of sync in this test.
```
[the remark this answers is not in the document]
Comment (arssher) [reply test_runner/batch_others/test_wal_acceptor.py:152]: looks ok

**Review by arssher — CHANGES_REQUESTED** [review]

File: test_runner/batch_others/test_wal_acceptor.py (hunk 2 above)
Comment (yeputons) [reply test_runner/batch_others/test_wal_acceptor.py:98]: Nice catch, thank you. It won't. Fixed.

**Review by arssher — APPROVED** [review]

**Pull request merged**